prov/bgq/src/fi_bgq_pmi.c

```diff
@@ -338,24 +341,29 @@ int PMI_KVS_Get (const char kvsname[], const char key[], char value[], int lengt
 		/* After the bgq node list loop write out last node blocks. */
 		if (current_node_id == bgq_node_list[0]) {
 			/* If only 1 node this is simple. */
-			node_block_write(value, current_node_block_value, current_block_id, 1, current_proc_count);
+			node_block_write(value, length, current_node_block_value, sizeof(current_node_block_value),
+					current_block_id, 1, current_proc_count);
 		}
 		else {
 			if (prev_proc_count == current_proc_count) {
 				/* Loop ended with last two nodes having equal number of procs. */
 				current_node_block_count++;
-				node_block_write(value, current_node_block_value, current_block_id, current_node_block_count, prev_proc_count);
+				node_block_write(value, length, current_node_block_value, sizeof(current_node_block_value),
+						current_block_id, current_node_block_count, prev_proc_count);
 			}
 			else {
 				/* Loop ended with last node having different number of procs than
 				   second-to-last node. */
-				node_block_write(value, current_node_block_value, current_block_id, current_node_block_count, prev_proc_count);
+				node_block_write(value, length, current_node_block_value, sizeof(current_node_block_value),
+						current_block_id, current_node_block_count, prev_proc_count);
 				current_block_id += current_node_block_count;
-				node_block_write(value, current_node_block_value, current_block_id, 1, current_proc_count);
+				node_block_write(value, length, current_node_block_value, sizeof(current_node_block_value),
+						current_block_id, 1, current_proc_count);
 			}
 		}
 		// add closing bracket to value
-		strcat(value,")");
+		int offset = strlen(value);
+		snprintf(value+offset, length-offset, ")");
```

Comment:
I would drop all changes to the bgq provider.  All development on this provider has been halted.

Reply:
I pulled out this commit.